lib/utils/class-alerts.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Automattic\VIP\Utils;
+
+class Alerts {
+	/**
+	 * The alerts service address
+	 *
+	 * @var string
+	 */
+	protected $service_address;
+
+	/**
+	 * The alerts service port
+	 *
+	 * @var string
+	 */
+	protected $service_port;
+
+	/**
+	 * Full path to the alerts service
+	 *
+	 * @var string
+	 */
+	protected $service_url;
+
+	/**
+	 * Instance of his Alerts class
+	 *
+	 * @var Alerts
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Contructor
+	 *
+	 * Set to protected to prevent direct instantiation.
+	 * Use Alerts::get_instance() to get an instance of Alerts
+	 */
+	protected function __construct() {
+		if ( ! defined( 'ALERT_SERVICE_ADDRESS' ) || ! ALERT_SERVICE_ADDRESS ) {
+			throw new Exception( 'Missing alerts service host configuration in ALERT_SERVICE_ADDRESS constant' );
+		}
+
+		if ( ! defined( 'ALERT_SERVICE_PORT' ) || ! ALERT_SERVICE_PORT ) {
+			throw new Exception( 'Missing alerts service port configuration in ALERT_SERVICE_PORT constant' );
+		}
+
+		$this->service_address = ALERT_SERVICE_ADDRESS;
+		$this->service_port = ALERT_SERVICE_PORT;
+		$this->service_url = sprintf( 'http://%s:%s/v1.0/alerts', $this->service_address, $this->service_port );
+	}
+
+	/**
+	 * Send the alert
+	 *
+	 * @param array $body The alert message body
+	 *
+	 * @return array Response details from wp_remote_post
+	 */
+	protected function send( $body ) {
+		$response = wp_remote_post( $this->service_url, [
+			'timeout' => 0.1,
+			'body' => json_encode( $body ),
+		] );
+
+		if ( is_wp_error( $response ) ) {
+			throw new Exception( sprintf( 'Error sending alert: %s', $response->get_error_message() ) );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Cache alert
+	 *
+	 * Used to throttle alerts requests
+	 *
+	 * @param $kind string Cache key
+	 * @param $interval integer When the cache data should expire, in seconds
+	 *
+	 * @return bool
+	 */
+	protected function add_cache( $key, $expire ) {
+		if ( function_exists( 'wp_cache_add' ) && function_exists( 'wp_cache_add_global_groups' ) ) {
+			wp_cache_add_global_groups( [ 'irc-ratelimit' ] );
+
+			return wp_cache_add( $key, 1, 'irc-ratelimit', $expire );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Validate IRC channel or user name
+	 *
+	 * @param $channel_or_user string
+	 *
+	 * @return string Validated and cleaned channel or user name
+	 */
+	protected function validate_channel_or_user( $channel_or_user ) {
```

Comment:
No benefit really. I'm not sure if it would make much of a difference since we will need an instance of `Alerts`  anyway in the main static functions like `chat()` and `opsgenie()`